textselect.cpp

```diff
@@ -48,8 +48,11 @@ static float substringSizeX(std::string_view s, size_t start, size_t length = st
     if (length == std::string_view::npos) stringEnd = s.end();
     else utf8::unchecked::advance(stringEnd, std::min(utf8Length(s), length));
 
+    // Dereferencing std::string_view::end() may be undefined behavior in some compilers,
+    // because of that, we need to get the pointer value manually if stringEnd == s.end().
+    const char* endPtr = stringEnd == s.end() ? s.data() + s.size() : &*stringEnd;

```

Comment:
The second parameter to `ImGui::CalcTextSize` is optional and `nullptr` by default, so when left unspecified, the function will continue until the end of the string is reached. Then, we won't need the `s.data() + s.size()` calculation and can take advantage of the default behavior:

```suggestion
    const char* endPtr = stringEnd == s.end() ? nullptr : &*stringEnd;
```

Reply:
I wonder if that's a good thing. Internally, when the second parameter is `nullptr` Dear ImGui does the same calculation but uses `strlen()` to figure it out the size of the string. Since we already have the size, having ImGui calculate the string size again is quite wasteful.
_Here's where ImGui calculates the end of the string: https://github.com/ocornut/imgui/blob/master/imgui_draw.cpp#L3914_

Of course, it's a small thing and won't be a problem unless strings are really, really large, so if you still prefer `nullptr` over `s.data() + s.size()` I have no other problems with it.

Reply:
That's a good point. Ideally, we want to minimize the amount of computations per frame. Like you said, this is only a small difference in the code so I'll merge it as it is.